### ISSUE TITLE
Rule proposal: prefer TypedArray's over `node:buffer`

### ISSUE THREAD

**Issue by jimmywarting** [sindresorhus/eslint-plugin-unicorn#1808]
### Description

This rule is more targeted for cross platform compatibility reasons. `node:buffer` is a node specific module and i mostly just see it as something that bloats browser bundles and runs slower. 

- `TextEncoder` / `TextDecoder` can be used instead to turn your buffer to / from string / Uint8Array and is also actually much faster then `Buffer.from(str)` (which originally use string_decoder) in the browser, 
- DataView can be used instead of reading / writing bytes
- base64 & Hex string should be discouraged b/c it takes up more bandwidth and isn't well suited for json or large data, there are better ways to send things in binary format like with FormData or just uploading raw data, it also waste processing time to encode / decode data and strings in v8 strings are [caped at 512 MiB](https://stackoverflow.com/a/61316641/1008999) now we have TypedArrays that are better
- `buffer.slice` is also overriding `uint8array.slice` with subarray, which is also unexpected and now also [deprecated ](https://nodejs.org/dist/latest-v17.x/docs/api/buffer.html#bufslicestart-end)
- the buffer userland polyfill for the browser is also lagging behind
- `Buffer.isBuffer` can be replaced with `instaceof Uint8Array` to be acceptable of both `Uint8Array` and Buffer's. something even better would be to accept any TypedArray using `ArrayBuffer.isView`
- TextDecoder is also able to handle BOM - unlike `buf.toString()`
- NodeJS internal modules have also already been more acceptable of accepting any TypedArray, where as it only supported Buffer initially and wouldn't accept a Uint8Array.

I bet if you read this [issue](https://github.com/nodejs/node/issues/41588) then you will maybe also be convinced that `node:buffer` is just unnecessary.

### Fail

```js
Buffer.from('hello world')
```

### Pass

```js
new TextEncoder().encode('hello world')
```

### Additional Info

Replacing `node:buffer` with `Uint8Array` can be a lot. and maybe not that easy to autofix
Maybe dividing it up to smaller task could be easier.

like 
- [ ] prefer `TextDecoder` over `buf.toString()`
- [ ] prefer `instanceof Uint8Array` over `Buffer.isBuffer` or something like that...
- [ ] prefer DataView over read/write methods

**Comment by sindresorhus** [comment]
I agree with the general idea, but it would be good if you could explicitly list more fail/pass cases.

**Comment by sindresorhus** [comment]
This is accepted in principle, but it needs more examples and a specific plan before we would accept a pull request.

**Comment by sindresorhus** [comment]
Closing as I plan some more narrow rules for this and the issue is not actionable.
